File: ruby/GameOfLife/src/position.rb
class Position
  attr_reader :x, :y

  # noinspection RubyInstanceVariableNamingConvention
  def initialize(x, y)
    @x = x
    @y = y
  end

  def eql?(other)
    other.x == @x && other.y == y
  end

  def hash
    @x * 31 ^ @y
  end
end

File: javascript/GameOfLife/js/position.js
class Position {
    constructor(x, y) {
        this._x = x;
        this._y = y;
    }

    equals(position) {
        return this._x == position.x &&
            this._y == position.y;
    }

    get x() {
        return this._x;
    }

    get y() {
        return this._y;
    }
}